server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -597,6 +584,17 @@ public void doRun()
           }
         }
     );
+    handoffNotifier.registerSegmentHandoffCallback(
+        new SegmentDescriptor(sink.getInterval(), sink.getVersion(), config.getShardSpec().getPartitionNum()),
+        mergeExecutor, new Runnable()
+        {
+          @Override
+          public void run()
+          {
+            abandonSegment(sink.getInterval().getStartMillis(), sink);
```

Comment:
It's possible that the sink is no longer in `sinks` at this point, so this runnable should check for that.